mesh/pncmesh.cpp

```diff
@@ -46,12 +46,38 @@ ParNCMesh::ParNCMesh(MPI_Comm comm, const NCMesh &ncmesh, int *part)
    // branches that only contain someone else's leaves (see Prune())
 }
 
+ParNCMesh::ParNCMesh(MPI_Comm comm, std::istream &input, int version,
+                     int &curved)
+   : NCMesh(input, version, curved)
+{
+   MyComm = comm;
+   MPI_Comm_size(MyComm, &NRanks);
+
+   int my_rank;
+   MPI_Comm_rank(MyComm, &my_rank);
+
+   int max_rank = 0;
+   for (int i = 0; i < leaf_elements.Size(); i++)
+   {
+      max_rank = std::max(elements[leaf_elements[i]].rank, max_rank);
+   }
```

Comment:
Shouldn't this be something like 
```cpp
bool owns_elements = true;
for (int i = 0; i < leaf_elements.Size(); i++)
{
    owns_elements &= elements[leaf_elements[i]].rank == my_rank;
}
```
? Or do I misunderstand the intent here?

Reply:
There is no need for a rank to own any elements ("empty" ranks are legal). This is just a simple sanity check when loading a parallel mesh: if there are elements (e.g. in the ghost layer) that are supposed to belong to a rank that is outside the range of the current communicator, it's clear the mesh was created on a larger MPI run and cannot be loaded.